app/views/RoomActionsView/index.js

```diff
@@ -271,6 +266,15 @@ class RoomActionsView extends React.Component {
 				testID: 'room-actions-auto-translate'
 			});
 		}
+		if (!isDirectRoom) {
```

Comment:
You can remove it.
Even if the user doesn't have access to pin, he always has access to see what's pinned.